harness/tests/experiment/pytorch/test_deepspeed_autotuning.py

```diff
@@ -65,55 +56,100 @@ def test_autotuning_module_transforms(
+    """
     model_dir = BASE_EXPERIMENT_FIXTURE_PATH.joinpath("example_experiment")
     config_path = model_dir.joinpath("autotune_config.yaml")
     submitted_config_dict = _utils.get_dict_from_yaml_or_json_path(config_path)
 
     with tempfile.TemporaryDirectory() as searcher_dir:
         searcher_dir = pathlib.Path(searcher_dir)
-        search_method = SimpleDSATSearchMethod(
+        search_method = search_method(
             submitted_config_dict=submitted_config_dict,
             model_dir=model_dir,
         )
-        mock_master_obj = MockMaster(
-            all_metrics=[MODEL_INFO_PROFILE_METRIC_FIXTURE, {"throughput": 1.0}]
-        )
+        mock_master_obj = MockMaster(all_metrics=all_metrics)
         search_runner = MockMasterSearchRunner(search_method, mock_master_obj, searcher_dir)
         search_runner.run(exp_config={}, context_dir="", includes=None)
-
-    # TODO: Use a more dynamic value if/when we enable users to configure this
-    exp_num_trials = _defaults.AUTOTUNING_DICT["tuner_num_trials"]
-    assert len(search_runner.state.trials_created) == exp_num_trials
-    assert len(search_runner.state.trials_closed) == exp_num_trials
-    assert len(search_runner.state.trial_progress) == exp_num_trials
-    # TODO: Handle the progress being 6 every time...
-    # for trial_uuid in search_runner.state.trial_progress:
-    #     assert(search_runner.state.trial_progress[trial_uuid] == 1.0)
-
-
-# TODO: Check that the trial tracker stops early when we trigger the early stopping criteria
-
-
-def test_random_search_method() -> None:
-    pass
-
-
-def test_trial_tracker() -> None:
-    pass
-
-
-def test_mbs_binary_search() -> None:
-    pass
+    return search_runner
+
+
+@pytest.mark.timeout(5)
+def test_deepspeed_autotune_happy_path() -> None:
+    """
+    Simulate the Deepspeed Autotune Search Methods end to end and make sure
+    nothing falls over
+    """
+    for search_method in _defaults.ALL_SEARCH_METHOD_CLASSES.values():
+        search_runner = _run_searcher(
+            search_method, [MODEL_INFO_PROFILE_METRIC_FIXTURE, {"throughput": 1.0}]
+        )
+        exp_num_trials = _defaults.AUTOTUNING_DICT["tuner_num_trials"]
+        assert len(search_runner.state.trials_created) == exp_num_trials
+        assert len(search_runner.state.trials_closed) == exp_num_trials
+        assert len(search_runner.state.trial_progress) == exp_num_trials
+        for trial_uuid in search_runner.state.trial_progress:
+            assert search_runner.state.trial_progress[trial_uuid] == 1.0
+
+
+@pytest.mark.timeout(5)
+def test_continuous_failures() -> None:
+    """Make sure that DSAT Search Methods can handle continuous failures"""
+    for search_method in _defaults.ALL_SEARCH_METHOD_CLASSES.values():
+        all_mock_metrics = [
+            MODEL_INFO_PROFILE_METRIC_FIXTURE,
+            {"throughput": 1.0},
+            {ERROR_METRIC_NAME: 1.0},
```

Comment:
A little confused about this test. When Trials fail, no metrics are sent back at all. Thought by the name that this is testing whether things go alright if every Trial fails.

It used to be that we would pass back the error message as a string, at least in some cases, but we stopped doing that a bit ago, in order to simplify. (Otherwise, Trials whose errors were caught and reported back and metrics would still go through `on_validation_completed` while those with errors which were not caught would go through `on_trial_exited_early` and the whole thing was a mess)

Reply:
So, in the case of the `ERROR_METRIC_NAME` metrics, we are taking that to mean "send a `trialExitEarly` instead of a `validationComplete`" it's not really sending a metric back, and is just a faked value for me to configure how the mock master should act for each of the trials

Reply:
Sounds good, maybe document that in a comment string?